asf_public_discourse_home_decarbonisation/utils/text_processing_utils.py

```diff
@@ -0,0 +1,205 @@
+"""
+Utils for processing text data.
+"""
+
+import re
+from nltk.corpus import stopwords
+from nltk import ngrams
+import nltk
+
+nltk.download("stopwords")
+from gensim.parsing.preprocessing import STOPWORDS
+import re
+import string
+from nltk.stem.wordnet import WordNetLemmatizer
+from nltk.tag import pos_tag
+
+from nltk import pos_tag
+from nltk.corpus import wordnet
+
+nltk.download("punkt")
+nltk.download("averaged_perceptron_tagger")
+nltk.download("wordnet")
+import pandas as pd
+from collections import Counter
+
+
+def remove_urls(text: str) -> str:
+    """
+    Removes URLs from text.
+    Args:
+        text: a string, tipically one or multiple sentences long
+    Returns:
+        text without URLs
+    """
+    pattern = r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
+    cleaned_text = re.sub(pattern, " ", text)
+    return cleaned_text
+
+
+def replace_punctuation_with_space(text: str) -> str:
+    """
+    Replaces punctuation with space.
+    Args:
+        text: a string, tipically one or multiple sentences long
+    Returns:
+        text wihtout punctuation
+    """
+    pattern = r"[{}]".format(re.escape(string.punctuation))
+    replaced_text = re.sub(pattern, " ", text)
+    replaced_text = re.sub(r"\s+", " ", replaced_text)  # Remove consecutive spaces
+    return replaced_text.strip()  # Remove leading/trailing spaces
+
+
+def remove_text_after_patterns(text: str) -> str:
+    """
+    Removes pattern of the form "xxx writes: ".
+
+    Args:
+        text (str): text to be cleaned
+
+    Returns:
+        str: cleaned text
+    """
+    # We use re.sub() to replace the pattern with an empty string
+    result = re.sub(r"\w+ wrote »", " ", text)
```

Comment:
good catch, thanks liz! this happens (and should happen) before any lemmatisation, so I just need to check which ones is the correct one. 
(In truth, we don't expect any matches to this because we're removing quotes from the text when extracting the data from the HTMLs, so having this here is just as a precaution)